### PR TITLE
6.4wip

### PR DESCRIPTION
**Pull request by jeffreytoffoli**
Updates for 6.4

### CODE REVIEW

File: portal-guide/walkthroughs/custom-text-walkthrough.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+![](../../images/browse_add.jpg)
+
+### Option 1: Browse your files
+
+You can upload your text directly from a `.csv` file. This means you can work with your favorite spreadsheet software or text editor when preparing your data for upload. Just use the provided "CSV template" to get started.
+
+Next, add your text data. At a minimum, you should add text to the `input.data.text.raw` field and the `input.data.concepts[i].id` field. Other fields are optional.
+
+* **Input.data.text.raw** - This is where you will input the text that you want to analyze with the awesome power of AI
+* **Input.data.concepts[i].id** - This is where you will list your custom concepts
```
Comment (zeiler) [inline portal-guide/walkthroughs/custom-text-walkthrough.md:26]: this isn't clear what to do here. What do I do with multiple concepts? I thought it was input.data.concepts[0].id as one column, input.data.concepts[1] as next column, etc. 

Also is there no input.data.concetpts[*].value columns to say 0 for negative and 1 for positive? 

Finally, the headers here are wrong, things won't work with `Input`, has to be `input`.

File: portal-guide/walkthroughs/visual-text-recognition-walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Visual Text Recognition Walkthrough
+
+Visual text recognition helps you understand text in images and videos.
+
+Visual text detection and recognition models need to be configured in a workflow so that you can detect text in your images.
```
Comment (zeiler) [inline portal-guide/walkthroughs/visual-text-recognition-walkthrough.md:5]: Same comment here as the API section, I don't understand from reading this what it is for or does.

File: api-guide/walkthroughs/visual-text-recognition-walkthrough.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    metadata=metadata
+)
+
+if post_workflows_response.status.code != status_code_pb2.SUCCESS:
+    raise Exception("Post workflows failed, status: " + post_workflows_response.status.description)
+```
+{% endtab %}
+
+{% tab title="gRPC Java" %}
+```java
```
[the remark this answers is not in the document]
Comment (jeffreytoffoli) [reply api-guide/walkthroughs/visual-text-recognition-walkthrough.md:75]: the other walkthrough is just python. this one was a quick iteration on other documentation, so getting it up in the other languages was quick. but, moving forward i'm planning on just python or node, per our conversation. Maybe we should just use Python for consistency?

File: api-guide/walkthroughs/text-model-predictions-walkthrough.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+
+## Python script for bulk Text model predictions
+```python
+from __future__ import absolute_import, division, print_function
+
+import argparse
+import os
+
+import pandas as pd
```
[the remark this answers is not in the document]
Comment (jeffreytoffoli) [reply api-guide/walkthroughs/text-model-predictions-walkthrough.md:46]: totally agree. This was based on some code that minh was working on. I will tweak it

File: api-guide/walkthroughs/visual-text-recognition-walkthrough.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (zeiler) [reply api-guide/walkthroughs/visual-text-recognition-walkthrough.md:75]: yeah it depends on the walkthrough. For server side definitely python as it's easier to read than node and other languages. For a web based example the new grpc web client that Rok is working on would be best so let's wait for that to be completed.

File: api-guide/walkthroughs/text-model-predictions-walkthrough.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeffreytoffoli) [reply api-guide/walkthroughs/text-model-predictions-walkthrough.md:46]: I think that we should just move this one to Zendesk

File: portal-guide/walkthroughs/custom-text-walkthrough.md (hunk 1 above)
Comment (jeffreytoffoli) [reply portal-guide/walkthroughs/custom-text-walkthrough.md:26]: ok, i think i have done a better job of explaining this, maybe take another look

File: api-guide/walkthroughs/text-model-predictions-walkthrough.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (zeiler) [reply api-guide/walkthroughs/text-model-predictions-walkthrough.md]: zendesk?

File: portal-guide/walkthroughs/custom-text-walkthrough.md [hunk 5]
```diff
@@ -14,16 +14,25 @@ Create a new application and select “Text” as your default workflow.
...
-![](../../images/browse_add.jpg)
-
 ### Option 1: Browse your files
 
 You can upload your text directly from a `.csv` file. This means you can work with your favorite spreadsheet software or text editor when preparing your data for upload. Just use the provided "CSV template" to get started.
 
-Next, add your text data. At a minimum, you should add text to the `input.data.text.raw` field and the `input.data.concepts[i].id` field. Other fields are optional.
+![](../../images/csv_template.jpg)
+
+Next, add your text data. At a minimum, you should add text to the `input.data.text.raw` field. You can add one concept per column to the `input.data.concepts[i].id` fields. For the `input.data.concepts[i].value` column, there are two options: enter the number `1` if the concept *is* present in the input, enter the value `0` if the concept is *not* present in the input (a negative example). If no value is entered, a default value of `1` will be assigned to your input.
```
Comment (zeiler) [inline portal-guide/walkthroughs/custom-text-walkthrough.md:23]: it's still now clear here what `[i]` is and sam with below for the table of fields.

**Pull request merged**